tilelang/autotuner/param.py

```
@@ -396,52 +397,74 @@ def _load_kernel_from_disk(
 
-        # save kernel
-        self._save_kernel_to_disk(path, self.kernel)
+            # save kernel
+            self._save_kernel_to_disk(staging_path, self.kernel, verbose)
+
+            # Atomic rename — directory becomes visible in one step.
+            os.rename(str(staging_path), str(path))
+        except OSError:
+            # Race: another process already created the final path.
+            shutil.rmtree(str(staging_path), ignore_errors=True)
+        except Exception:
```

Ruff (0.15.7)

[warning] Do not catch blind exception: `Exception`

(BLE001)